lib/cocoapods/validator.rb

```diff
@@ -612,7 +612,7 @@ def configure_pod_targets(targets, target_installation_results, deployment_targe
                             else
                               pod_target.spec_swift_versions.map(&:to_s).find do |v|
                                 v == derived_swift_version
-                              end || pod_target.target_definition_swift_version
+                              end || pod_target.swift_version
```

Comment:
this will internally check if the pod has a swift version set in its spec otherwise return the one from the target definition.

Reply:
and this was the regression from 1.7.1 -> 1.7.2